### DISCUSSION TITLE
Error position when animation cancal

### DISCUSSION THREAD

**Discussion by Coloryr** [AvaloniaUI/Avalonia#13302] [Q&A]
It is my animation
```
double end = to.Bounds.Width;
var animation = new Animation
{
    FillMode = FillMode.Forward,
    Easing = SlideEasing,
    Children =
    {
        new KeyFrame
        {
            Setters =
            {
                new Setter
                {
                    Property = TranslateTransform.XProperty,
                    Value = Mirror ? end - 1 : -end + 1
                }
            },
            Cue = new Cue(0d)
        },
        new KeyFrame
        {
            Setters =
            {
                new Setter
                {
                    Property = TranslateTransform.XProperty,
                    Value = 0
                }
            },
            Cue = new Cue(1d)
        }
    },
    Duration = Duration
};
animation.RunAsync(to, cancellationToken);
```

start it,
```
App.SidePageSlide300.Start(null, DockPanel1, _cancel1.Token);
```

and cancel in runring,

```
_cancel1.Cancel();
```

When start in next time, it show in error pos.

![GIF](https://github.com/AvaloniaUI/Avalonia/assets/26276037/fb7c1c5f-9fec-4581-9b99-3574a57303bf)

**Comment by Coloryr** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/b68a2769-32cf-4d59-86fb-5a793af1f618)
`Bounds.Width` is a fixed value.

**Reply by thevortexcloud** [reply]
Bounds is not a fixed value per se. It's the actual physical size of the control. It can change, but is not guaranteed to be. Typically it happens when the window is resized. Beyond that I can't say what the problem here is.

**Reply by Coloryr** [reply]
Oh, thanks.

**Reply by Coloryr** [reply]
Which value should be used at this animation?

**Comment by Coloryr** [comment]
```
_cancel1.Cancel();
_cancel1.Dispose();
_cancel1 = new();
StackPanel1.IsVisible = true;
Dispatcher.UIThread.Post(() =>
{
    App.SidePageSlide300.Start(null, DockPanel1, _cancel1.Token);
});
```
use this code, no work.
![GIF](https://github.com/AvaloniaUI/Avalonia/assets/26276037/687d1847-3bb5-4aec-be3e-27b94c014c74)

Correct position.
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/b0d7343d-46e4-4c2b-b991-44e8a2dd4a7d)

Wrong position.
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/7580d121-fc10-437d-aa32-6134c71fe75d)

Is look like TranslateTransform.XProperty not reset?

**Comment by Coloryr** [comment]
When animation cancel, the control not reset the pos.
![image](https://github.com/AvaloniaUI/Avalonia/assets/26276037/3d60faab-54f8-48c1-add0-a9c287de3a4c)

**Comment by Coloryr** [comment]
Is look like `RenderTransform` not reset.

this code work.
```
else if (e.PropertyName == "SideOpen")
{
    _cancel1.Dispose();
    _cancel1 = new();
    
    StackPanel1.IsVisible = true;
    Dispatcher.UIThread.Post(() =>
    {
        _ = App.SidePageSlide300.Start(null, DockPanel1, _cancel1.Token);
    });
}
else if (e.PropertyName == "SideClose")
{
    _cancel1.Cancel();
    _cancel1.Dispose();
    _cancel1 = new();
    await App.SidePageSlide300.Start(DockPanel1, null, _cancel1.Token);
    StackPanel1.IsVisible = false;
}
```

![GIF](https://github.com/AvaloniaUI/Avalonia/assets/26276037/63d63392-9984-4729-abed-eefae17fbf47)

**Comment by kyuranger** [comment]
Btw, honestly Avalonia animation system has so many strange bugs all the time. And I suggest declaring the animation by XAML and invoking it by adding classes in code-behind.
